worker/healthcheck/worker_test.go

```diff
@@ -0,0 +1,115 @@
+package healthcheck_test
+
+import (
```

Comment:
Hey reviewer, this should give a good description of the possible scenarios that the checker might run through.

Reply:
Hey @cscosta, I would be in favor of setting a grace period in garden for the containers, especially because we're hoping to do that for check containers as part of #3079. I assume this PR will be merged first, so it will allow us to see if there are any unforeseen repercussions of not having all containers tracked by GC.

Reply:
Hey,

I gave a try at using grace periods for garden containers and baggageclaim
volumes, and I really enjoy the idea! It makes things much simpler and
pushing the responsibility to those that are managing the "repositories"
of containers and volumes seems to be very good.

While `garden`'s `grace_period` indeed does what I expected (reap the
containers after such period), it seems like baggeclaim's TTL is not really
like that - it keeps resetting the expiration time you set, and performing
some kind of heartbeating that I don't have much context on.

Is there any way that I can achieve `grace_period` for baggageclaim in the
codebase *as is* today?

Thanks!

Reply:
@cirocosta For volumes with a TTL, you have to call `.Release` in order to stop the heartbeating:

https://github.com/concourse/baggageclaim/blob/572a539e53765714d0b76ac412fac2e53bbb6017/client.go#L98

Reply:
hmmmm I see 🤔 As is, that's something that seems to not be supported by the API baggageclaim exposes though: https://github.com/concourse/baggageclaim/blob/572a539e53765714d0b76ac412fac2e53bbb6017/routes.go

If we'd be exposing a `Release` in the API just for such purpose, do you think it'd be a good idea to instead move the its API forward to have a `grace_time` instead and get rid of that healthchecking logic in Baggageclaim? (assuming it's really not used anywhere - is it?)

Wdyt?

thx!

Reply:
@cirocosta we are planning on making a baggageclaim plugin for garden so we can make ephemeral check containers with TTLs. This means whenever you create a container on garden, it will volumize on baggageclaim. When the grace period expires, garden will delete both container and volumes it knows about (in this case call baggageclaim delete).

Would it be fine for us to have a single Garden health probe that indirectly checks baggageclaim as well?

Reply:
Oh, interesting, do you have an issue where I can keep track of that work? I guess that's a bunch of work, so, not something I could rely on very soon, right?

It'd definitely be fine, the only thing I'd like to avoid is having extra API calls for setting TTLs or anything like that as it'd mean that we could potentially leave things behind not being GCed somehow.

thx!

Reply:
We're planning to spike on it this week. It's a part of #3079. Essentially this plugin is a cli that we point garden to with create and destroy cmds. The cli wraps `baggageclaim.client` so a lot of the hardwork is already done. I hope its quick, I'll keep you in the loop.

Reply:
Coooool, thaanks!